metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -639,7 +639,10 @@ def generate_usage_statistics(self, project_id: str) -> Iterable[MetadataWorkUni
         tables: Dict[str, List[str]] = {}
```

Comment:
unrelated to these changes but we should use a `defaultdict` here to simplify the code below

Reply:
@treff7es  - the reason I said this was that the code below does a check of `if tables[dataset]` assign, otherwise extend. By using a defaultdict(list), you can remove the `if` and always extend

More a code cleanup thing - not required for this PR